frontend/src/containers/SignupView.tsx

```diff
@@ -1,69 +1,108 @@
 import { Form, Input, Icon, Button } from 'antd';
 import { FormComponentProps } from "antd/lib/form";
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { connect } from 'react-redux';
 import { RouteComponentProps, NavLink } from 'react-router-dom';
 
 import { authSignup } from "actions/auth";
+import { AuthError } from "reducers/auth";
 import { State } from "reducers/index";
 
 interface StateToProps {
   loading: boolean;
-  error: Error | null;
+  error: AuthError | null;
 }
 
 interface DispatchToProps {
   authSignup: (username: string, email: string, password1: string, password2: string) => void;
 }
 
-type Props = RouteComponentProps & StateToProps & DispatchToProps & FormComponentProps;
+interface ErrorProps {
+  error: AuthError;
+};
 
+const RegistrationError = ({ error }: ErrorProps) => {
+  if (error.response.status !== 400) {
+    return <div>{error.message}</div>;
+  }
+  const parsedErrors = [];
+  let key = 0;
+  for (const errors of Object.values(error.response.data)) {
+    for (const error of errors) {
+      parsedErrors.push(<li key={key}>{error}</li>);
+      key++;
+    }
+  }
+  return (
+    <div>
+      <ul>
+        {parsedErrors}
+      </ul>
+    </div>
+  )
+}
 
+type Props = RouteComponentProps & StateToProps & DispatchToProps & FormComponentProps;
 
-class RegistrationForm extends React.Component<Props> {
-  state = {
-    confirmDirty: false,
-  };
+const RegistrationForm = (props: Props) => {
+  const [submitted, setSubmitted] = useState(false);
+  const [error, setError] = useState<AuthError | null>(null);
 
-  handleSubmit = (e: React.FormEvent) => {
+  useEffect(() => {
+    if (props.loading) { return; }
+    const error = submitted && (props.error != null);
+    if (props.error !== null) {
+      setError(props.error);
+    }
+    if (!error && submitted) {
+      props.history.push('/login');
+    }
+    if (submitted) {
+      setSubmitted(false);
+    }
+  }, [props.loading, props.error, props.history, error, submitted]);
```

Comment:
Have you also considered props destructuring at the top of the component? It's makes component more readable imo.